.github/workflows/main.yml

```diff
@@ -10,39 +10,35 @@ on:
 jobs:
   cypress-run:
     runs-on: ubuntu-latest
-    container: cypress/browsers:node16.14.0-slim-chrome99-ff97
+    container: cypress/browsers:latest
     steps:
       - name: Checkout
         uses: actions/checkout@v2
 
-      - name: install pip
-        run: |
-          apt update -qq
-          apt install -y python3-pip
-          pip3 --version
-
       - name: Checkout xblock-sdk repo
         uses: actions/checkout@v2
         with:
           repository: openedx/xblock-sdk
           path: xblock-sdk
-
-      - name: Setup xblock-sdk
-        working-directory: xblock-sdk
+      - uses: actions/setup-python@v5
+        id: setup_python
+        with:
+          python-version: '3.12'
+      - name: Setup python virtual env
         run: |
-          pwd
-          pip install -qr requirements/dev.txt
-          pip install -qr requirements/local.txt
-
-      - name: Setup dependencies
+          python -m venv .venv
+      - name: activate venv
+        run: .venv/bin/activate
```

Comment:
> Since each run command in GitHub Actions initiates a new shell instance, activating a virtual environment in one step does not carry over to subsequent steps.
I don't think this is how it works. We are installing xblock-sdk in one step and then installing h5p xblock in next step. The step where we install h5 xblock should break if its being install in new env because it has Xblock package as dependency.

Reply:
I have referred to documentation  [here](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idsteps) , in here it does say "because steps run in their own process, changes to environment variables are not preserved between steps."

Reply:
As per my research, I understand the concern about the dependencies between the installation of xblock-sdk and h5p xblock. Indeed, if each step were completely isolated, the installation of h5p xblock would fail due to missing dependencies installed in previous steps. However, while each run command does initiate a new shell instance, the changes to the filesystem (like installed packages in the virtual environment) persist between steps.However, changes to the shell environment itself, such as environment variables set by activating a virtual environment, do not persist. This is why directly using the Python executable from the virtual environment, as opposed to activating it, ensures that the correct environment and dependencies are consistently utilized across different steps.

However it's not considered not the best practice to rely on activating virtual environments in GitHub Actions for each command due to the statelessness of each run command's shell environment. Directly using the Python executable from the virtual environment's path in each command is more reliable and aligns better with GitHub Actions' execution model.This approach avoids potential issues with environment persistence and ensures that each command is executed with the correct environment setup.